DllImportGenerator/DllImportGenerator/MarshallingAttributeInfo.cs

```diff
@@ -366,6 +368,29 @@ CountInfo CreateCountInfo(AttributeData marshalUsingData, ImmutableHashSet<strin
             return NoCountInfo.Instance;
         }
 
+        private TypePositionInfo? CreateForParamIndex(AttributeData attrData, int paramIndex, ImmutableHashSet<string> inspectedElements)
```

Comment:
Seems like we are updating `inspectedElements`, can we pass this as `ref`?

Reply:
We actually aren't updating `inspectedElements`. It's an `ImmutableHashSet<T>`, so calling `Add()` creates a new instance. It makes it much easier to handle unwrapping the recursion-detection state as we come back up the stack, since it is handled automatically.